### ISSUE TITLE
`spack uninstall -f` is broken

### ISSUE THREAD

**Issue by citibeth** [LLNL/spack#858]
@alalazo  It looks like PR #670 might have broken "spack uninstall -f".  Can you please look into it?

```
[me@ankeli icebin]$ spack uninstall -f openblas
==> The following packages will be uninstalled : 


==> Do you want to proceed ? [y/n]
y
[me@ankeli icebin]$ ls /home/rpfische/spack/opt/spack/linux-x86_64/gcc-4.9.3/openblas-0.2.17-lvi5fks7zompzclvc2lyzwilfefal4ld/lib
blas.a  cmake  libblas.a  liblapack.a  libopenblas.a  libopenblas_nehalemp-r0.2.17.a
```

**Comment by alalazo** [comment]
Sure.

**Comment by alalazo** [comment]
##### TLDR
I cannot reproduce the problem, so it's hard for me to look into that with a debugger. From the output you posted it seems that your query `openblas` didn't produce any matching installed specs. At first glance this may be due to one of these two calls gone wrong  :
```python
specs = spack.cmd.parse_specs(args.packages)
uninstall_list = concretize_specs(specs, args.all, args.force)
...
```
The second one involves a query to the db, so... can you re-try after a `spack reindex` to see if this solves the issue?

##### Details
Output I get when trying to uninstall a package **associated with multiple installed specs**:
```bash
$ spack uninstall -f  szip
==> Error: szip matches multiple packages:

-- production / clang@3.8.0 -------------------------------------
uldzrgd szip@2.1

-- production / gcc@4.8 -----------------------------------------
35xe6vt szip@2.1

-- production / gcc@5.3.0 ---------------------------------------
a3rdlo5 szip@2.1

==> Error: You can either:
    a) Use a more specific spec, or
    b) use spack uninstall -a to uninstall ALL matching specs.
```
Output I get when trying to uninstall a package **associated with a single installed spec**:
```bash
$ spack uninstall -f  szip%clang
==> The following packages will be uninstalled : 

-- production / clang@3.8.0 -------------------------------------
uldzrgd szip@2.1

==> Do you want to proceed ? [y/n]
```
Output I get when trying to uninstall a package **not associated with an installed spec**:
```bash
$ spack uninstall -f thisdoesnotexist
==> The following packages will be uninstalled : 


==> Do you want to proceed ? [y/n]
```
I wouldn't say the last output is wrong, but surely it can be confusing. @citibeth , @tgamblin would you prefer a different message in case no packages to are found by querying the db ?

**Comment by citibeth** [comment]
I will make a clean dev-branch Spack on CentOS 7 and post the results of my
test.

**Comment by alalazo** [comment]
@citibeth In any case I think the message to the user should be improved if nothing matches the query. What's there is really tricky, in particular considering that you're asked permission to proceed on doing nothing...

**Comment by alalazo** [comment]
@citibeth Any news on this issue?

**Comment by tgamblin** [comment]
@citibeth: can you still reproduce this?

**Comment by citibeth** [comment]
With 6mo hindsight, my guess is this was some database problem that would have been fixed with `spack reindex`.  The only thing I might suggest is... if nothing is found that matches, don't ask the user for permission, and say something like:

```
$ spack uninstall -f thisdoesnotexist
==> No packages were found to uninstall.  Doing nothing.
```

**Comment by alalazo** [comment]
Ok PR incoming...
